future-of-mc-bootstrap/Readme.md

```diff
@@ -0,0 +1,57 @@
+# Future of Management Cluster bootstrap and Creation
+
+
+## Background
+
+With the work of openstack team rocket started to work on a new way to create
+management clusters. Goals were
+- Full automation with as less interactive steps as possible
+- Use CAPI clusters from the start
+- Use Kubernetes native tooling instead of terraform and virtual machines
+- Create a reliable way to recreate the management cluster from scratch if it
+  is messed up by our tests
+- Have a freshly created management cluster every beginning of the week to avoid
+  configuration drift and manual changes sneaking in.
+
+## Current state
+
+- Implemented with bash and make.
+- Automation still involves some manual steps for secret management and
+  last pass access.
+- With increased usage and accelerated growth of multiple teams using it some
+  issues surfaced
+  - No automated tests and CI, so that it is easy to mess it up for the other
+    team
+  - Bash / Make is not appreciated as a tool of choice by a some people
+  - Secret management is hard to fully automate in its current starte
+  - Credential creation is hard to automate as well (OIDC, Github, etc)
+  - Manual CI by recreating test MCs on a weekly basis does not scale.
+
+## Suggesstions to move on
+
+### Short Term
+
+- Separate build targets for different providers
+- Work on automated tests even if they are not perfect
+- Only keep SOPS encryption key and shared secrets in lastpass
+- Move all infrastructure configuration and installation secrets into a separate
+  repository encrypted with SOPS
+- Use kubectl apply to get started, move to flux as soon as we have a clear picture
```

Comment:
where we use kubectl apply? do you mean clusterctl init?

Reply:
Use kubectl apply to get the SOPS encrypted secrets to the newly created managment cluster until a flux setup that can do it.

Reply:
This is also what @kopiczko diagram illustrates.

Reply:
@kopiczko I included the diagram as SVG into the document. The SVG includes the original drawio diagram, so that it can be edited in the future.